modules/http/http-curl-compression.c

```diff
@@ -87,4 +98,62 @@ int _zstd_string(GString *compressed, const GString *message)
 {
   //not yet implemented
   g_assert_not_reached();
+}
+
+int _deflate_type_compression(GString *compressed, const GString *message, const gint deflate_algorithm_type)
```

Comment:
Please add unit tests.

Reply:
I had to start working on an other syslog-ng-related project, but these are coming SOON™. End-to-end tests are also in the pipeline. In the meantime, we've done manual testing with @MrAnno.